src/main/io/ledstrip.c

```diff
@@ -898,26 +898,26 @@ static void applyLedThrustRingLayer(bool updateNow, timeUs_t *timer)
 
 static void applyRainbowLayer(bool updateNow, timeUs_t *timer)
 {
+    //use offset as a fixed point number
     static int offset = 0;
 
     if (updateNow) {
-        *timer += HZ_TO_US(ledStripConfig()->ledstrip_rainbow_freq);
+        offset += ledStripConfig()->ledstrip_rainbow_freq;
+        *timer += HZ_TO_US(100);
```

Comment:
Let's introduce a macro for the LED task rate (in `ledstrip.h`) so that there are no magic numbers floating around in the codebase.
You also need to replace the number  with the macro in `tasks.c - line 411`
```suggestion
        *timer += HZ_TO_US(TASK_LEDSTRIP_RATE_HZ);
```